src/test/protocol/STValidation_test.cpp

```diff
@@ -38,6 +38,16 @@ class STValidation_test : public beast::unit_test::suite
     {
         testcase ("Deserialization");
 
+        constexpr unsigned char payload0[] =
+        { // specifies a secp256k1 public key
+            0x72, 0x00, 0x73, 0x21, 0x03, 0x76, 0x08, 0x7D, 0x78, 0x33, 0x5F, 0xE2, 0x08, 0xEE, 0xE1, 0x5D,
+            0x30, 0x96, 0x50, 0xFA, 0xAE, 0x60, 0xF4, 0xC5, 0xB7, 0xF3, 0x62, 0xAF, 0x97, 0x43, 0xF1, 0x72,
+            0x01, 0x9C, 0xBB, 0x20, 0xC8, 0x76, 0x31, 0x30, 0x37, 0x5f, 0x5f, 0x63, 0x6c, 0x61, 0x73, 0x73,
+            0x5f, 0x74, 0x79, 0x70, 0x65, 0x5f, 0x69, 0x6e, 0x66, 0x6f, 0x45, 0x00, 0xe6, 0x88, 0x54, 0x72,
+            0x75, 0x73, 0x74, 0x53, 0x65, 0x74, 0x65, 0x61, 0x74, 0x65, 0x88, 0x00, 0xe6, 0x88, 0x00, 0xe6,
+            0x73, 0x00, 0x72, 0x00, 0x8a, 0x00, 0x88, 0x00, 0xe6
+        };
+
```

Comment:
Consider moving into the scope where it is used and renaming to `payload`. I recommend the same for the other payloads. Nit- keep at 80 cols.

Reply:
Considering condensing to something like the following
```
        const std::vector<std::vector<unsigned char>> payload = {
            // specifies a secp256k1 public key
            {0x72, 0x00, 0x73, 0x21, 0x03, 0x76, 0x08, 0x7D, 0x78, 0x33,
             0x5F, 0xE2, 0x08, 0xEE, 0xE1, 0x5D, 0x30, 0x96, 0x50, 0xFA,
             0xAE, 0x60, 0xF4, 0xC5, 0xB7, 0xF3, 0x62, 0xAF, 0x97, 0x43,
             0xF1, 0x72, 0x01, 0x9C, 0xBB, 0x20, 0xC8, 0x76, 0x31, 0x30,
             0x37, 0x5f, 0x5f, 0x63, 0x6c, 0x61, 0x73, 0x73, 0x5f, 0x74,
             0x79, 0x70, 0x65, 0x5f, 0x69, 0x6e, 0x66, 0x6f, 0x45, 0x00,
             0xe6, 0x88, 0x54, 0x72, 0x75, 0x73, 0x74, 0x53, 0x65, 0x74,
             0x65, 0x61, 0x74, 0x65, 0x88, 0x00, 0xe6, 0x88, 0x00, 0xe6,
             0x73, 0x00, 0x72, 0x00, 0x8a, 0x00, 0x88, 0x00, 0xe6},
            // specifies an Ed25519 public key
            {0x72, 0x00, 0x73, 0x21, 0xed, 0x78, 0x00, 0xe6, 0x73, 0x00,
             0x72, 0x00, 0x3c, 0x00, 0x00, 0x00, 0x88, 0x00, 0xe6, 0x73,
             0x38, 0x00, 0x00, 0x8a, 0x00, 0x88, 0x4e, 0x31, 0x30, 0x5f,
             0x5f, 0x63, 0x78, 0x78, 0x61, 0x62, 0x69, 0x76, 0x31, 0x30,
             0x37, 0x5f, 0x5f, 0x63, 0x6c, 0x61, 0x73, 0x73, 0x5f, 0x74,
             0x79, 0x70, 0x65, 0x5f, 0x69, 0x6e, 0x66, 0x6f, 0x45, 0x00,
             0xe6, 0x88, 0x54, 0x72, 0x75, 0x73, 0x74, 0x53, 0x65, 0x74,
             0x65, 0x61, 0x74, 0x65, 0x88, 0x00, 0xe6, 0x88, 0x00, 0xe6,
             0x73, 0x00, 0x72, 0x00, 0x8a, 0x00, 0x88, 0x00, 0xe6},
            // specifies an Ed25519 public key
            {0x73, 0x21, 0xed, 0xff, 0x03, 0x1c, 0xbe, 0x65, 0x22,
             0x61, 0x9c, 0x5e, 0x13, 0x12, 0x00, 0x3b, 0x43, 0x00,
             0x00, 0x00, 0xf7, 0x3f, 0x3f, 0x3f, 0x3f, 0x3f, 0x3f,
             0x3f, 0x3f, 0x13, 0x13, 0x13, 0x3a, 0x27, 0xff},
            // Has no public key at all
            {0x72, 0x00, 0x76, 0x31, 0x30, 0x37, 0x5f, 0x5f, 0x63, 0x6c, 0x61,
             0x73, 0x73, 0x5f, 0x74, 0x79, 0x70, 0x65, 0x5f, 0x69, 0x6e, 0x66,
             0x6f, 0x45, 0x00, 0xe6, 0x88, 0x54, 0x72, 0x75, 0x73, 0x74, 0x53,
             0x65, 0x74, 0x65, 0x61, 0x74, 0x65, 0x88, 0x00, 0xe6, 0x88, 0x00,
             0xe6, 0x73, 0x00, 0x72, 0x00, 0x8a, 0x00, 0x88, 0x00, 0xe6}
        };

        try
        {
            for (auto const& p : payload)
            {
                SerialIter sit{p.data(), p.size()};
                std::make_shared<ripple::STValidation>(
                    sit,
                    [](PublicKey const& pk) {return calcNodeID(pk);},
                    false);
            }
            fail("An exception should have been thrown");
        }
        catch (std::exception const& e)
        {
            BEAST_EXPECT(
                strcmp(e.what(), "Invalid public key in validation") == 0);
        }
```